### ISSUE TITLE
No local vimrc file has been sourced with pseudopaths like fugitive://<clone>/.git/<object>/<path>

### ISSUE THREAD

**Issue by avimmer** [embear/vim-localvimrc#89]
This is more of a feature request. Recently while working with a revision of a file with fugitive I found that while the filetype is set properly no settings, maps or augroups from my local vimrc had been added to the buffer. I had some language specific stuff there which might have been of help.

The reason why no local vimrc had been sourced is obviously that the file name as per vim does not exist
- it's an URI with `fugitive:` scheme
- the path described with an object id does not exist in the `.git` tree

Here's an example of debug output I had
```
== START LocalVimRC() ============================
running for file "fugitive:///Users/REDACTED/whatever/.git//c90ab82d9aa69597829d851c08593f3642f3c1da/internal/benchmark_test.go"
found files: [{'resolved': 'fugitive:/.lvimrc', 'unresolved': 'fugitive:///.lvimrc'}, {'resolved': 'fugitive:/Users/.lvimrc', 'unresolved': 'fugitive:///Users/.lvimrc'}, {'resolved': 'ugi tive:/Users/REDACTED/.lvimrc', 'unresolved': 'fugitive:///Users/REDACTED/.lvimrc'}, {'resolved': 'fugitive:/Users/REDACTED/whatever/.lvimrc', 'unresolved': 'fugitive:///Users/REDACTED/whatever/.lvimrc'}, {'resolved': 'fugitive:/Users/REDACTED/whatever/.git/.lvimrc', 'unresolved': 'fugitive:///Users/REDACTED/whatever/.git//.lvimrc'}, {'resolved': 'fugitive:/Users/REDACTED/whatever/.git/c90ab82d9aa69597829d851c08593f3642f3c1da/.lvimrc', 'unresolved': 'fugitive:///Users/REDACTED/whatever/.git//c90ab82d9aa69597829d851c08593f3642f3c1da/.lvimrc'}, {'resolved': 'fugitive:/Users/REDACTED/whatever/.git/c90ab82d9aa69597829d851c08593f3642f3c1da/internal/.lvimrc', 'unresolved': 'fugitive:///Users/REDACTED/whatever/.git//c90ab82d9aa69597829d851c08593f3642f3c1da/internal/.lvimrc'}]
candidate files: [{'resolved': 'fugitive:/.lvimrc', 'unresolved': 'fugitive:///.lvimrc'}, {'resolved': 'fugitive:/Users/.lvimrc', 'unresolved': 'fugitive:///Users/.lvimrc'}, {'resolved': ' fugitive:/Users/REDACTED/.lvimrc', 'unresolved': 'fugitive:///Users/REDACTED/.lvimrc'}, {'resolved': 'fugitive:/Users/REDACTED/whatever/.lvimrc', 'unresolved': 'fugitive:///Users/REDACTED/whatever/.lvimrc'}, {'resolved': 'fugitive:/Users/REDACTED/whatever/.git/.lvimrc', 'unresolved': 'fugitive:///Users/REDACTED/whatever/.git//.lvimrc'}, {'resolved': 'fugitive:/Users/REDACTED/whatever/.git/c90ab82d9aa69597829d851c08593f3642f3c1da/.lvimrc', 'unresolved': 'fugitive:///Users/REDACTED/whatever/.git//c90ab82d9aa69597829d851c08593f3642f3c1da/.lvimrc'}, {'resolved': 'fugitive:/Users/REDACTED/whatever/.git/c90ab82d9aa69597829d851c08593f3642f3c1da/internal/.lvimrc', 'unresolved': 'fugitive:///Users/REDACTED/whatever/.git//c90ab82d9aa69597829d851c08593f3642f3c1da/internal/.lvimrc'}]
== END LocalVimRC() ==============================
```
In this example `/Users/REDACTED/whatever/.git//c90ab82d9aa69597829d851c08593f3642f3c1da` is not a real path, but something projected by fugitive.

I had pwd set to `/Users/REDACTED/whatever` and was expecting some way of loading my local vimrc located at `/Users/REDACTED` or even in `whatever` directory if I had one there since nothing at all could have been loaded. I worked around this by setting the file name to an arbitrary value, like `foo`, which made vim-localvimrc believe it's an unsaved buffer somewhere down the pwd. After that I issued :LocalVimRC and the file finally sourced. I find this suboptimal however as it's not that obvious and requires making sure you will not switch the window to another buffer. In some cases you are losing contents that way.

I would appreciate an optional argument to :LocalVimRC command which would tell the plugin where should it start traversing for local vimrcs. Like
- `:LocalVimRC .` would start pwd
- `:LocalVimRC foo` from pwd
- `:LocalVimRC ../foo` from ..
basically the same behaviour as for the file name. Other option could be a setting which tell what the plugin should fall back to in case resolving things will file name failed. Maybe you could think of some even better option.

Appreciate your work!
Have a nice day

**Comment by avimmer** [comment]
> Could you please try the attached patch and give me feedback if that suits your needs?

Pretty much it does. Thanks!

**Comment by embear** [comment]
Added above change in commit ebb73832e6795967e5a52db3636a37282871b218
